contracts/src/TeleporterTokenDestination.sol

```diff
@@ -347,18 +385,23 @@ abstract contract TeleporterTokenDestination is
             "TeleporterTokenDestination: invalid token source address"
         );
         BridgeMessage memory bridgeMessage = abi.decode(message, (BridgeMessage));
-        uint256 scaledAmount = scaleTokens(bridgeMessage.amount, true);
+
+        // If the contract was not previously known to be collateralized, it is now given that
+        // the source has sent a message to mint funds.
+        if (!isCollateralized) {
+            isCollateralized = true;
+        }
```

Comment:
I think the source contract is the source of truth for keeping track of a destination's collateral, and since adding collateral happens all on the source side, without the destination being involved directly, prefer to leave as is without destination calculating its own collateral